### PR TITLE
Make windows runners output object be consistent with other runner output objects

### PR DESCRIPTION
**Pull request by Kami**
This pull request updates a result / output dictionary returned by the Windows runners to match local and remote runner output object.

The change is breaking / partially backward incompatible so it needs to be included in minor / major release (2.7.0) and we need to document that change in changelog and upgrade notes document.

## TODO

- [x] Update affected tests
- [x] New tests
- [x] Changelog entry
- [x] Upgrade notes entry - https://github.com/StackStorm/st2docs/pull/707

Related issue #4044.

### CODE REVIEW

File: contrib/runners/windows_runner/windows_runner/windows_command_runner.py [hunk 1]
```diff
@@ -93,13 +93,12 @@ def run(self, action_parameters):
...
             error = self._parse_winexe_error(stdout=stdout, stderr=stderr)
 
-        result = stdout
-
         output = {
             'stdout': stdout,
             'stderr': stderr,
-            'exit_code': exit_code,
-            'result': result
+            'return_code': exit_code,
```
Comment (Kami) [inline contrib/runners/windows_runner/windows_runner/windows_command_runner.py]: I personally prefer exit code over return code, but I made the change for the consistency reasons.

**Comment by Kami** [comment]
While working on this, I also noticed windows runner doesn't correctly handle / use "timeout" live action / execution status so I fixed that (windows runner probably predates "timeout" status).

File: contrib/runners/windows_runner/windows_runner/windows_command_runner.py (hunk 1 above)
Comment (LindsayHill) [reply contrib/runners/windows_runner/windows_runner/windows_command_runner.py]: Agreed - I think that consistency here is more helpful

**Comment by LindsayHill** [comment]
Looking good so far. I think that you're right about some of this predating other changes. I suspect we updated the Python & Command runners, but didn't make the same changes to these runners. 

Assuming we move to pywinrm at some point in the near future, hopefully most of these changes will still be relevant.

**Comment by Kami** [comment]
@LindsayHill this PR is ready for review now.

**Pull request merged**